spectrocloud/resource_cloud_account_maas.go

```diff
@@ -0,0 +1,129 @@
+package spectrocloud
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/spectrocloud/hapi/models"
+	"github.com/spectrocloud/terraform-provider-spectrocloud/pkg/client"
+)
+
+func resourceCloudAccountMaas() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceCloudAccountMaasCreate,
+		ReadContext:   resourceCloudAccountMaasRead,
+		UpdateContext: resourceCloudAccountMaasUpdate,
+		DeleteContext: resourceCloudAccountMaasDelete,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"maas_api_endpoint": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"maas_api_key": {
+				Type:      schema.TypeString,
+				Optional:  true,
+				Sensitive: true,
+			},
+		},
+	}
+}
+
+func resourceCloudAccountMaasCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c := m.(*client.V1alpha1Client)
+
+	// Warning or errors can be collected in a slice type
+	var diags diag.Diagnostics
+
+	account := toMaasAccount(d)
+
+	uid, err := c.CreateCloudAccountMaas(account)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(uid)
+
+	resourceCloudAccountMaasRead(ctx, d, m)
```

Comment:
shouldn't we be return diags?

Reply:
@saamalik I checked this method we always return diags.